repose-aggregator/core/repose-core/src/main/java/org/openrepose/nodeservice/response/LocationHeaderBuilder.java

```diff
@@ -56,6 +60,24 @@ public static void setLocationHeader(HttpServletRequest originalRequest, HttpSer
         }
     }
 
+    private static URL getProxiedHostUrl(String targetHost) {
+        URL returnUrl = null;
+        URI targetUri = null;
+        try {
+            targetUri = new URI(targetHost);
+        } catch (URISyntaxException e) {
+            LOG.error("Invalid target host url: " + targetHost, e);
+        }
+        if (targetUri != null && targetUri.getScheme() != null && targetUri.getHost() != null) {
+            try {
+                returnUrl = new URL(targetUri.getScheme(), targetUri.getHost(), targetUri.getPort(), "");
+            } catch (MalformedURLException ex) {
+                LOG.error("Invalid host url: " + targetUri, ex);
```

Comment:
This should be a `WARN` at most since processing is able to continue.

Reply:
On it.